File: 15_FullStack/Exam_03.06.2024/frontendreact/src/Components/Models/Task.ts
import { Person } from "./Person";

export class Task {
    public id:number;
    public name:string;
    public responsible:Person;
    public scheduledDate:Date;
    public completed:boolean;

    constructor(id:number,name:string,responsible:Person,scheduledDate:Date,completed:boolean){
        this.id = id;
        this.name = name;
        this.responsible = responsible;
        this.scheduledDate = scheduledDate;
        this.completed = completed;
    }
}

File: 15_FullStack/Exam_03.06.2024/BackEnd_Spring/src/main/java/com/example/BackEnd_Spring/Beans/Task.java
package com.example.BackEnd_Spring.Beans;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Column(nullable = false)
    private String name;
    @OneToOne
    private Person responsible;
    @Column(nullable = false)
    private LocalDate scheduledDate;
    private boolean isCompleted;
}
